Library/Homebrew/cmd/update-report.rb

```diff
@@ -250,10 +250,23 @@ def migrate_tap_migration
       tabs = dir.subdirs.map { |d| Tab.for_keg(Keg.new(d)) }
       next unless tabs.first.tap == tap # skip if installed formula is not from this tap.
       new_tap = Tap.fetch(new_tap_name)
-      new_tap.install unless new_tap.installed?
-      # update tap for each Tab
-      tabs.each { |tab| tab.tap = new_tap }
-      tabs.each(&:write)
+      # For formulae migrated to cask: Auto-install cask or provide install instructions.
+      if new_tap_name == "caskroom/cask"
+        if new_tap.installed? && (HOMEBREW_REPOSITORY/"Caskroom").directory?
+          ohai "#{name} has been moved to Homebrew Cask. Installing #{name}..."
+          system "brew", "cask", "install", name
+        else
+          ohai <<-EOS.undent
+            #{name} has been moved to Homebrew Cask. To install it run:
+              brew cask install #{name}
+          EOS
```

Comment:
`ohai` isn't suitable for multi-line output, not to mention that it truncates its output. This snippet in `brew irb`:

```ruby
name = "abi-compliance-checker"
ohai <<-EOS.undent
  #{name} has been moved to Homebrew Cask. To install it run:
    brew cask install #{name}
EOS
```

Produces the following output on my 120-column wide terminal (notice the truncation):

```
==> abi-compliance-checker has been moved to Homebrew Cask. To install it run:
  brew cask install abi-compliance-checke
```

I'd either switch to a simple `puts` or (if the highlighting is really important) use something like:

```ruby
ohai "#{name} has been moved to Homebrew Cask.", <<-EOS.undent
  To install it run:
    brew cask install #{name}
EOS
```